libpod/container_internal_linux.go

```diff
@@ -615,6 +615,8 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 	}
 
 	if c.config.Systemd {
+		// https://systemd.io/CONTAINER_INTERFACE/#environment-variables
+		g.AddProcessEnv("container_uuid", c.ID()[:32])
```

Comment:
You should check if container_uid environment variable was set by the user and not set it here.  Only set if the user did not set it.